src/Peachpie.CodeAnalysis/FlowAnalysis/Passes/DiagnosticWalker.cs

```diff
@@ -103,13 +109,56 @@ protected override void VisitCFGInternal(ControlFlowGraph x)
 
             base.VisitCFGInternal(x);
 
+            CheckParams();
+
+            if (!_callsParentCtor && _routine.Name == Devsense.PHP.Syntax.Name.SpecialMethodNames.Construct.Value
+                && _routine.ContainingType.BaseType?.ResolvePhpCtor() != null)
+            {
+                // Missing calling parent::__construct
+                _diagnostics.Add(_routine, _routine.SyntaxSignature.Span.ToTextSpan(), ErrorCode.WRN_ParentCtorNotCalled, _routine.ContainingType.Name);
+            }
+
             // analyse missing or redefined labels
             CheckLabels(x.Labels);
 
             // report unreachable blocks
             CheckUnreachableCode(x);
         }
 
+        private void CheckParams()
+        {
+            // Check the compatibility of type hints with PhpDoc, if both exist
+            if (_routine.PHPDocBlock != null)
+            {
+                for (int i = 0; i < _routine.SourceParameters.Length; i++)
+                {
+                    var param = _routine.SourceParameters[i];
+
+                    // Consider only parameters passed by value, with both typehints and PHPDoc comments
+                    if (!param.Syntax.IsOut && !param.Syntax.PassedByRef
+                        && param.Syntax.TypeHint != null
+                        && param.PHPDocOpt != null && param.PHPDocOpt.TypeNamesArray.Length != 0)
+                    {
+                        var hintType = param.Type;
+                        TypeSymbol docType = null;
+
+                        var tmask = PHPDoc.GetTypeMask(TypeCtx, param.PHPDocOpt.TypeNamesArray, _routine.GetNamingContext());
+                        if (!tmask.IsVoid && !tmask.IsAnyType)
+                        {
+                            docType = DeclaringCompilation.GetTypeFromTypeRef(TypeCtx, tmask);
+                        }
+
+                        if (!docType?.IsOfType(hintType) ?? false)
```

Comment:
isn't it easier just to check `!docType.IsOfType(hintType)` right on line 149 :) All these `Nullable` and `??` and false ... is just an overkill

Reply:
It is :-) I just didn't realize that ```GetTypeFromTypeRef``` cannot return ```null```. Fixed in 2eee6f3ae66e8a265d3704329c9789ab05bc34bd.